### PR TITLE
feat(RadioCard): Create component

### LINKED ISSUE #982: Radio Button Enhanced
### Description
When designing forms in our exemplar project, we have replaced radio buttons with buttons in a number of cases. This makes the selection easier and clearer to make.

From the button set, only one button can be activated (equivalent of selecting a radio button).

radio-button-tile

### PR DESCRIPTION
**Pull request by m7kvqbe1**
## Related issue

Closes #982 

## Overview

Implement RadioCard component.

## Reason

>The Radio Button Enhanced is a rich-content variation of the Radio Button. It provides a way to include more than just a label for a radio selection. The original designs have been modified to include a Title and a Radio button icon - this is important to demonstrate to the user that the radio buttons form part of a group and work with behaviour they're used to.

## Work carried out

- [x] Implement RadioCard component
- [x] Add automated tests
- [x] Add developer documentation
- [ ] Add design documentation

## Screenshot

Screenshot 2020-06-25 at 11 38 37
Screenshot 2020-06-25 at 12 15 33

## Developer notes

Kept RadioCard as an 'uncontrolled' component similar to the standard Radio version.

### CODE REVIEW

File: packages/css-framework/src/components/_radio-card.scss [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+  border-radius: 4px;
+  border: 1px solid f.color("neutral", "100");
+  outline: none;
+
+  &:hover {
+    border-color: f.color("action", "400");
+    box-shadow: 0 0 0 3px f.color("action", "100");
+    cursor: pointer;
+
+    .rn-radio input ~ .rn-radio__checkmark {
```
Comment (m7kvqbe1) [inline packages/css-framework/src/components/_radio-card.scss:21]: I attempted to programmatically simulate a hover on the child radio via a ref.

Along the lines of this:

```typescript
  const handleMouseEnter = (_: React.MouseEvent<HTMLDivElement>) => {
    ref.current.dispatchEvent(new MouseEvent('mouseenter'))
  }
```

 It wouldn't work for some reason. 

As such i've ended up duplicating a couple of hover styles here instead.

File: packages/react-component-library/src/components/RadioCard/RadioCard.test.tsx [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    }
+  })
+
+  describe('when a field has no errors', () => {
+    beforeEach(() => {
+      form.errors = {}
+    })
+
+    describe('and the field has a label', () => {
+      beforeEach(() => {
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RadioCard/RadioCard.test.tsx:84]: I find this level of `beforeEach` nesting hard to read. I appreciate it has been copied from another test but I feel we should tidy these things up as we get chance.

File: packages/react-component-library/src/components/RadioCard/RadioCard.tsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+  const handleKeyUp = (_: React.KeyboardEvent) => {
+    ref.current.focus()
+  }
+
+  return (
+    <div
+      className="rn-radio-card"
+      role="radio"
+      onClick={handleClick}
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RadioCard/RadioCard.tsx:34]: When removing these two lines (33-34), the tests still pass. It feels like we need something that actually covers the fact that this `div` is clicked and the `radio` acts appropriately.

File: packages/react-component-library/src/components/RadioCard/RadioCard.tsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+  title,
+  description,
+  tabIndex = 0,
+  isChecked,
+  ...rest
+}) => {
+  const ref = useRef(null)
+
+  const handleClick = (_: React.MouseEvent<HTMLDivElement>) => {
+    ref.current.click()
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RadioCard/RadioCard.tsx:22]: I can see this line is not covered, see comment about lines 33-34.

File: packages/react-component-library/src/components/RadioCard/RadioCard.tsx [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  const handleKeyUp = (_: React.KeyboardEvent) => {
+    ref.current.focus()
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RadioCard/RadioCard.tsx:26]: I can see this line is not covered, see comment about lines 33-34.

File: packages/react-component-library/src/components/RadioCard/RadioCard.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  }
+
+  return (
+    <div
```
Comment (thyhjwb6) [inline packages/react-component-library/src/components/RadioCard/RadioCard.tsx:30]: I have been trying to keep the attributes alphabetical. Seems to be easier to read.

File: packages/docs-site/src/library/pages/components/form/radio-card.md [hunk 7]
```diff
@@ -0,0 +1,173 @@
...
+    Required: 'True',
+    Default: '',
+    Description: 'Can be manually provided or sent from Formik. Indicates any errors and which fields have been touched',
+  },
+  {
+    Name: 'id',
+    Type: 'string',
+    Required: 'False',
+    Default: 'unique id',
+    Description: 'Provide the ID for a field and tie it to its label. If one is not provided then a unique id will be created',
```
Comment (thyhjwb6) [inline packages/docs-site/src/library/pages/components/form/radio-card.md]: Maybe capitalise `id`

**Review by thyhjwb6 — COMMENTED** [review]
Works well, some comments around a gap in testing mostly.

File: packages/css-framework/src/components/_radio-card.scss [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+  }
+}
+
+.rn-radio-card__description {
+  color: f.color("neutral", "400");
+  padding-top: f.spacing("4");
+}
+
+.rn-radio-card .rn-radio__label {
+  font-size: f.font-size("l");
```
Comment (hxltrhuxze) [inline packages/css-framework/src/components/_radio-card.scss]: Can we change this to `font-size("m")`? It's correct here based on the designs, however on reflection it looks better with a slightly reduced title size. I'll update the Sketch file to match too.

**Review by thyhjwb6 — APPROVED** [review]
👍

**Pull request merged**